standard-integration/server.js

```diff
@@ -1,34 +1,149 @@
+      // "PayPal-Mock-Response": '{"mock_application_codes": "INTERNAL_SERVER_ERROR"}'
+    },
+    method: 'POST',
+    body: JSON.stringify(payload),
+  });
+
+  return handleResponse(response);
+};
+
+/**
+ * Capture payment for the created order to complete the transaction.
+ * @see https://developer.paypal.com/docs/api/orders/v2/#orders_capture
+ */
+const captureOrder = async (orderID) => {
+  const accessToken = await generateAccessToken();
+  const url = `${base}/v2/checkout/orders/${orderID}/capture`;
+
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${accessToken}`,
+      // Uncomment one of these to force an error for negative testing (in sandbox mode only). Documentation:
+      // https://developer.paypal.com/tools/sandbox/negative-testing/request-headers/
+      // "PayPal-Mock-Response": '{"mock_application_codes": "INSTRUMENT_DECLINED"}'
+      // "PayPal-Mock-Response": '{"mock_application_codes": "TRANSACTION_REFUSED"}'
+      // "PayPal-Mock-Response": '{"mock_application_codes": "INTERNAL_SERVER_ERROR"}'
+    },
+  });
+
+  return handleResponse(response);
+};
+
+async function handleResponse(response) {
   try {
-    const order = await paypal.createOrder();
-    res.json(order);
+    const jsonResponse = await response.json();
+    return {
+      jsonResponse,
+      httpStatusCode: response.status,
+    };
   } catch (err) {
-    res.status(500).send(err.message);
+    const errorMessage = await response.text();
+    throw new Error(errorMessage);
+  }
+}
+
+app.post('/api/orders', async (req, res) => {
+  try {
+    // use the cart information passed from the front-end to calculate the order amount detals
+    const { cart } = req.body;
+    const { jsonResponse, httpStatusCode } = await createOrder(cart);
+    res.status(httpStatusCode).json(jsonResponse);
+  } catch (error) {
+    console.error('Failed to create order:', error);
+    res.status(500).json({ error: 'Failed to create order.' });
   }
 });
 
-app.post("/my-server/capture-paypal-order", async (req, res) => {
-  const { orderID } = req.body;
+app.post('/api/orders/:orderID/capture', async (req, res) => {
   try {
-    const captureData = await paypal.capturePayment(orderID);
-    res.json(captureData);
-  } catch (err) {
-    res.status(500).send(err.message);
+    const { orderID } = req.params;
+    const { jsonResponse, httpStatusCode } = await captureOrder(orderID);
+    res.status(httpStatusCode).json(jsonResponse);
+  } catch (error) {
+    console.error('Failed to create order:', error);
+    res.status(500).json({ error: 'Failed to capture order.' });
   }
 });
 
-app.listen(PORT, () => {
-  console.log(`Server listening at http://localhost:${PORT}/`);
+// serve index.html
+app.get('/', (req, res) => {
+  res.sendFile(path.resolve('./index.html'));
+});
+
+app.listen(8888, () => {
+  console.log('listening on http://localhost:8888/');
```

Comment:
option to use a `PORT` env var here that defaults to 8888

Reply:
Great suggestion @jshawl! I just pushed a commit to optionally read PORT from an env and default to 8888